beacon_node/beacon_chain/src/kzg_utils.rs

```diff
@@ -1,54 +1,57 @@
 use kzg::{Error as KzgError, Kzg, BYTES_PER_BLOB};
-use types::{Blob, BlobsSidecar, EthSpec, Hash256, KzgCommitment, KzgProof, Slot};
+use types::{Blob, EthSpec, KzgCommitment, KzgProof};
 
+/// Converts a blob ssz List object to an array to be used with the kzg
+/// crypto library.
 fn ssz_blob_to_crypto_blob<T: EthSpec>(blob: Blob<T>) -> kzg::Blob {
     let blob_vec: Vec<u8> = blob.into();
     let mut arr = [0; BYTES_PER_BLOB];
     arr.copy_from_slice(&blob_vec);
     arr.into()
 }
 
-pub fn validate_blobs_sidecar<T: EthSpec>(
+/// Validate a single blob-commitment-proof triplet from a `BlobSidecar`.
+pub fn validate_blob_sidecar<T: EthSpec>(
     kzg: &Kzg,
-    slot: Slot,
-    beacon_block_root: Hash256,
-    expected_kzg_commitments: &[KzgCommitment],
-    blobs_sidecar: &BlobsSidecar<T>,
+    blob: Blob<T>,
+    kzg_commitment: KzgCommitment,
+    kzg_proof: KzgProof,
 ) -> Result<bool, KzgError> {
-    if slot != blobs_sidecar.beacon_block_slot
-        || beacon_block_root != blobs_sidecar.beacon_block_root
-        || blobs_sidecar.blobs.len() != expected_kzg_commitments.len()
-    {
-        return Ok(false);
-    }
-
-    let blobs = blobs_sidecar
-        .blobs
-        .clone() // TODO(pawan): avoid this clone
-        .into_iter()
-        .map(|blob| ssz_blob_to_crypto_blob::<T>(blob))
-        .collect::<Vec<_>>();
-
-    kzg.verify_aggregate_kzg_proof(
-        &blobs,
-        expected_kzg_commitments,
-        blobs_sidecar.kzg_aggregated_proof,
+    kzg.verify_blob_kzg_proof(
+        ssz_blob_to_crypto_blob::<T>(blob),
+        kzg_commitment,
+        kzg_proof,
     )
 }
 
-pub fn compute_aggregate_kzg_proof<T: EthSpec>(
+/// Validate a batch of blob-commitment-proof triplets from multiple `BlobSidecars`.
+pub fn validate_blobs_sidecar<T: EthSpec>(
```

Comment:
nit: 

```suggestion
pub fn validate_blob_sidecars<T: EthSpec>(
```

Reply:
Changed it to `validate_blobs` to match the fork choice spec.